src/libraries/System.Security.Cryptography.Encoding/tests/Cbor/CborWriter.cs

```diff
@@ -25,38 +26,109 @@ public partial class CborWriter : IDisposable
+        private HashSet<(int Offset, int Length)>? _keyEncodingRanges = null; // all key encoding ranges up to encoding equality
 
-        public CborWriter(CborConformanceLevel conformanceLevel = CborConformanceLevel.Lax, bool encodeIndefiniteLengths = false, bool allowMultipleRootLevelValues = false)
+        /// <summary>
+        ///   The <see cref="CborConformanceLevel"/> used by this writer.
+        /// </summary>
+        public CborConformanceLevel ConformanceLevel { get; }
+
+        /// <summary>
+        ///   True if the writer automatically converts indefinite-length encodings into definite-length equivalents.
+        /// </summary>
+        public bool ConvertIndefiniteLengthEncodings { get; }
+
+        /// <summary>
+        ///   Declares whether this writer allows multiple root-level CBOR data items.
+        /// </summary>
+        public bool AllowMultipleRootLevelValues { get; }
+
+        /// <summary>
+        ///   Gets the writer's current level of nestedness in the CBOR document.
+        /// </summary>
+        public int CurrentDepth => _nestedDataItems is null ? 0 : _nestedDataItems.Count;
+
+        /// <summary>
+        ///   Gets the total number of bytes that have been written to the buffer.
+        /// </summary>
+        public int BytesWritten => _offset;
+
+        // Returns true iff a complete CBOR document has been written to buffer
+
+        /// <summary>
+        ///   True if the writer has completed writing a complete root-level CBOR document,
+        ///   or sequence of root-level CBOR documents.
+        /// </summary>
+        public bool IsWriteCompleted => _currentMajorType is null && _itemsWritten > 0;
+
+        /// <summary>
+        ///   Create a new <see cref="CborWriter"/> instance with given configuration.
+        /// </summary>
+        /// <param name="conformanceLevel">
+        ///   Specifies a <see cref="CborConformanceLevel"/> guiding the conformance checks performed on the encoded data.
+        ///   Defaults to <see cref="CborConformanceLevel.Lax" /> conformance level.
+        /// </param>
+        /// <param name="convertIndefiniteLengthEncodings">
+        ///   Enables automatically converting indefinite-length encodings into definite-length equivalents.
+        ///   Allows use of indefinite-length write APIs in conformance levels that otherwise do not permit it.
+        ///   Defaults to <see langword="false" />.
+        /// </param>
+        /// <param name="allowMultipleRootLevelValues">
+        ///   Specify if multiple root-level values are to be supported by the writer.
+        ///   When set to <see langword="false" />, the reader will throw an <see cref="InvalidOperationException"/>
+        ///   if trying to write beyond the scope of one root-level CBOR data item.
+        /// </param>
+        /// <exception cref="ArgumentOutOfRangeException">
+        ///   <paramref name="conformanceLevel"/> is not defined.
```

Comment:
"Not defined" feels a little... soft.  Like someone could believe it included "I called the constructor as `new CborWriter()`".

```suggestion
        ///   <paramref name="conformanceLevel"/> is not a defined <see cref="CborConformanceLevel"/>.
```